xarray/namedarray/core.py

```diff
@@ -855,6 +861,249 @@ def _to_dense(self) -> NamedArray[Any, _DType_co]:
+        data = self._data.transpose(
+            axes
+        )  # TODO: replace this with use array-api-compat function
+        return self._replace(dims=dims, data=data)
+
+    @property
+    def T(self) -> NamedArray[Any, _DType_co]:
+        """Return a new object with transposed dimensions."""
+        if self.ndim != 2:
+            raise ValueError(
+                f"x.T requires x to have 2 dimensions, got {self.ndim}. Use x.permute_dims() to permute dimensions."
+            )
+
+        data = self._data[::-1, ::-1]
+        dims = self.dims[::-1]
+        return self._replace(dims=dims, data=data)
+
+    def _check_dims(self, dims: _DimsLike | Mapping[_Dim, int]) -> None:
+        if isinstance(dims, dict):
+            dims_keys = dims
+        else:
+            dims_keys = dims if isinstance(dims, str) else list(dims)
+        if missing_dims := set(self.dims) - set(dims_keys):
+            raise ValueError(
+                f"new dimensions {dims!r} must be a superset of "
+                f"existing dimensions {self.dims!r}. missing dims: {missing_dims}"
+            )
+
+    def broadcast_to(
+        self, dim: _DimsLike | Mapping[_Dim, int] | None = None, **dim_kwargs: Any
+    ) -> NamedArray[Any, _DType_co]:
+        """
+        Broadcast the NamedArray to a new shape by extending its dimensions.
+
+        This method allows for the expansion of the array's dimensions to a specified shape.
+        New dimensions with specified sizes can be added, and existing dimensions can be resized.
+        It handles both positional and keyword arguments for specifying the dimensions to broadcast.
+
+        Parameters
+        ----------
+        dim : dict, str, sequence of str, optional
+            Dimensions to broadcast the array to. If a dict, keys are dimension names and values are the new sizes.
+            If a string or sequence of strings, new dimensions are added with a size of 1.
+
+        **dim_kwargs : Any
+            Additional dimensions specified as keyword arguments. Each keyword argument specifies the name of the new dimension and its size.
+
+        Returns
+        -------
+        NamedArray
+            A new NamedArray with the broadcasted dimensions.
+
+        Examples
+        --------
+        >>> data = np.asarray([[1.0, 2.0], [3.0, 4.0]])
+        >>> array = xr.NamedArray(("x", "y"), data)
+        >>> array.sizes
+        {'x': 2, 'y': 2}
+
+        >>> broadcasted = array.expand_dims("lat").broadcast_to(x=2, y=2, lat=6)
+        >>> broadcasted.sizes
+        {'lat': 6, 'x': 2, 'y': 2}
+        """
+
+        from xarray.core import duck_array_ops
+
+        combined_dims = either_dict_or_kwargs(dim, dim_kwargs, "broadcast_to")
+
+        # check that the dimensions are valid
+        self._check_dims(list(combined_dims.keys()))
+
+        # create a dictionary of the current dimensions and their sizes
+        current_shape = dict(zip(self.dims, self._data.shape))
```

Comment:
```suggestion
        # create a dictionary of the current dimensions and their sizes
        current_shape = self.sizes
```

Don't we have `NamedArray.sizes`?